docs/proposals/object-search.md

```diff
@@ -0,0 +1,160 @@
+Updates are categorized below along with the possible actions that the downstream controller can take:
+
+| Update Type                    | Action                                       |
+|-------------------------------|---------------------------------------------- |
+|New Object-search-service Created             |Sends the task information to LCs.|
+|Object-search-service Deleted                 | The controller sends the delete event to LCs.|
+
+### Upstream Controller
+![](./images/object-search-upstream-controller.png)
+
+The upstream controller watches for object-search-service task updates from the edge node and applies these updates against the API server in the cloud.
+Updates are categorized below along with the possible actions that the upstream controller can take:
+
+| Update Type                        | Action                                        |
+|-------------------------------     |---------------------------------------------- |
+|Object-search-service Reported State Updated    |  The controller appends the reported status of the object-search-service in the cloud. |
+
+
+### Details of api between GM(cloud) and LC(edge)
+1. GM(downstream controller) syncs the task info to LC:
+    ```go
+    // POST <namespace>/sedna/downstream/objectsearchservices/<name>/insert
+    // body same to the task crd of k8s api, omitted here.
+    ```
+
+1. LC uploads the task status which reported by the worker to GM(upstream controller):
+    ```go
+    // POST <namespace>/sedna/upstream/objectsearchservices/<name>/status
+       
+    // ObjectSearchServiceStatus defines status that send to GlobalManager
+    type ObjectSearchServiceStatus struct {
+    	Phase  string  `json:"phase"`
+    	Status string  `json:"status"`
+    	Output *Output `json:"output"`
+    }
+    
+    // Output defines task output information
+    type Output struct {
+    	Models   []Model   `json:"models"`
+    	TaskInfo *TaskInfo `json:"taskInfo"`
+    }
+    
+    // Model defines the model information
+    type Model struct {
+    	Format string `json:"format"`
+    	URL    string `json:"url"`
+    }
+    
+    // TaskInfo defines the task information
+    type TaskInfo struct {
+    	InferenceNumber   int     `json:"inferenceNumber"`
+    	StartTime         string  `json:"startTime"`
+    	CurrentTime       string  `json:"currentTime"`
+    }
+
+    ```
+
+### Flow of object search service creation
+- The flow of object search service creation:
+
+![](./images/object-search-flow-creation.png)
+The object search service controller watches the creation of object search service crd in the cloud, syncs them to lc via the cloudhub-to-edgehub channel, and creates the inference workers on the edge nodes specified by the user.
+These inference workers are started by the kubeedge at the edge nodes.
+
+## Workers Communication
+![](./images/object-search-worker-communication.png)
```

Comment:
this worker communication is too simple, need to add more details.

Reply:
The worker communication figures are updated.